### ISSUE TITLE
Masking scores before Softmax in Attention discussed here - https://www.tensorflow.org/tutorials/text/nmt_with_attention

### ISSUE THREAD

**Issue by UtpalMattoo** [tensorflow/tensorflow#43941]
Hi - 

In the Attention mechanism here: [Neural Machine Translation](https://www.tensorflow.org/tutorials/text/nmt_with_attention) - see related code included below (from `class BahdanauAttention`), we did not mask the scores before applying softmax, so that we only sum the attention distribution to 1 for only the actual sequence length, just like it is discussed here: [masking scores before Softmax](https://github.com/juditacs/snippets/blob/master/deep_learning/masked_softmax.ipynb)

Is there a reason why we did not mask by setting previously padded elements to '-inf' so that their Softmax score would be 0?

Thanks!

`

    # score shape == (batch_size, max_length, 1)
    # we get 1 at the last axis because we are applying score to self.V
    # the shape of the tensor before applying self.V is (batch_size, max_length, units)
    score = self.V(tf.nn.tanh(
        self.W1(query_with_time_axis) + self.W2(values)))

    # attention_weights shape == (batch_size, max_length, 1)
    attention_weights = tf.nn.softmax(score, axis=1)

    # context_vector shape after sum == (batch_size, hidden_size)
    context_vector = attention_weights * values
    context_vector = tf.reduce_sum(context_vector, axis=1)

    return context_vector, attention_weights`

**Comment by tilakrayal** [comment]
Is this still an issue?
Could you please update TensorFlow to the latest stable version v.2.6 and let us know if you are facing the same error. Thanks!
